packages/lodestar/src/api/impl/lodestar/index.ts

```diff
@@ -181,9 +187,52 @@ export function getLodestarApi({
         data: network.discv5?.kadValues().map((enr) => enr.encodeTxt()) ?? [],
       };
     },
+
+    async getLiveness(indices: ValidatorIndex[], epoch: Epoch): Promise<{data: LivenessResponseData[]}> {
+      if (indices.length === 0) {
+        return {
+          data: [],
+        };
+      }
+      const state = await chain.getHeadStateAtCurrentEpoch();
+      const epochCtx = state.epochCtx;
+      if (epoch < epochCtx.previousShuffling.epoch || epoch > epochCtx.nextShuffling.epoch) {
+        throw new Error(
+          `Request epoch ${epoch} is more than one epoch previous or after from the current epoch ${epochCtx.currentShuffling.epoch}`
+        );
+      }
+
+      return {
+        data: indices.map((index: ValidatorIndex) => {
+          return {
+            index,
+            epoch,
+            isLive: isLive(chain, index, epoch),
+          };
+        }),
+      };
+    },
   };
 }
 
+function isLive(chain: IBeaconChain, index: ValidatorIndex, epoch: Epoch): boolean {
+  const startSlot = computeStartSlotAtEpoch(epoch);
+  const endSlot = computeEndSlotForEpoch(epoch);
+
+  let proposedBlock = false;
+  for (let slot = startSlot; slot <= endSlot; slot++) {
+    if (chain.seenBlockProposers.isKnown(slot, index)) {
+      proposedBlock = true;
+      break;
+    }
+  }
+
+  const hasAggregated = chain.seenAggregators.isKnown(epoch, index);
```

Comment:
You should be very careful of false negatives. The `seenAggregators` cache prunes every epoch so I contains limited data.
```ts
  isKnown(targetEpoch: Epoch, validatorIndex: ValidatorIndex): boolean {
    return this.validatorIndexesByEpoch.get(targetEpoch)?.has(validatorIndex) === true;
  }
```
What we really want (in pseudo code) is:
```ts
  isKnown(targetEpoch: Epoch, validatorIndex: ValidatorIndex): boolean {
    const validatorIndexes = this.validatorIndexesByEpoch.get(targetEpoch)
    if (validatorIndexes) {
      return validatorIndexes.has(validatorIndex)
    } else {
      throw "=( i don't have this data anymore, not sure if seen or not"
    }
  }
```